crypto/pem/pem_lib.c

```diff
@@ -63,7 +63,7 @@ int PEM_def_callback(char *buf, int num, int rwflag, void *userdata)
         memset(buf, 0, (unsigned int)num);
         return -1;
     }
-    return strlen(buf);
+    return strnlen(buf, (size_t)num);
```

Comment:
Unfortunately, `strnlen()` is a GNUism, i.e. non-standard.  C11 defines a similar function, `strnlen_s`, but OpenSSL requires C89/C90, so we'd really need to have an OpenSSL-specific fallback function.

Reply:
OPENSSL_strnlen() ?

Reply:
> OPENSSL_strnlen() ?

Fixed